Improve theme remixing UI

LeaVerou opened the pull request:
Part of [#574](https://github.com/shoelace-style/webawesome/issues/574).

This PR implements:

- Permalinks (closes #717)
- Previews of colors, palettes, typography (closes #673)
- Change theme remixing icon to `arrows-rotate`
- Use `<wa-details>` to reduce clutter (open by default if theme is remixed)
- Style tweaks
- Prototype of brand color picker 

It’s not perfect, but I think it's much better than the current UI.

Future work:
- When you select a theme for colors and the palette is set to its default, it should change to that theme's default palette, but that doesn't quite work. No idea why.
- Eventually we may want to show actual font names in the typography cards, but we currently don't have this data

This is quite affected by #653 so I’m hoping we could prioritize it.

Screenshots:

When visiting a (regular) theme page:
<img width="1185" alt="image" src="https://github.com/user-attachments/assets/79f0b03c-3143-44f6-aac5-1b22b67afbd7" />

Hovered question icon:
<img width="354" alt="image" src="https://github.com/user-attachments/assets/42908347-8c7d-432c-a1fa-dcca01682691" />

Colors selector:
<img width="927" alt="image" src="https://github.com/user-attachments/assets/1b0b6bdc-718d-433a-8258-210fbf224c11" />

Palette picker: (will be significantly improved once #696 is merged):
<img width="1170" alt="image" src="https://github.com/user-attachments/assets/9d105529-f9ca-410d-b77e-6efb33f33f5f" />

Typography picker:
<img width="839" alt="image" src="https://github.com/user-attachments/assets/145736f9-eb86-49a1-bc25-00d9d782c022" />

LeaVerou committed:
Improve theme remixing UI

- Permalinks (closes #717)
- Previews of colors, palettes, typography (closes #673)
- Change theme remixing icon to `arrows-rotate`
- Use `<wa-details>` to reduce clutter (open by default if theme is remixed)
- Style tweaks

lindsaym-fa commented:
Great ideas!

I can't seem to get anything to work on the Vercel preview, and some of the same issues occur intermittently locally. 😿 Here's what I'm seeing in Safari and Chrome:

1) The Remix selections always claim to default to _Default_, _Anodized_, and _Default_, no matter which theme I start with. (Vercel only; haven't reproduced locally.)
<img width="971" alt="Screenshot 2025-02-06 at 5 30 19 PM" src="https://github.com/user-attachments/assets/1c005761-0fa9-4962-8c2f-1c81c6a3681b" />

---

2) Changing the Remix selections doesn't update the theme preview. (Always using Vercel; intermittently locally)

Things initially work fine locally, but I can consistently get myself into this broken state by visiting one theme page, changing any Remix selection, then navigating to another theme page. At that point, no selections change the theme preview on any page. Refreshing resolves the issue.
https://github.com/user-attachments/assets/3b058756-8e14-4043-8dfe-f67287c3fc1d

---

3) When I **refresh** the page, all values in the Remix dropdowns appear to be selected. This isn't the case if I navigate to the page anew. (Vercel only; haven't reproduced locally.)
<img width="1356" alt="Screenshot 2025-02-06 at 5 24 16 PM" src="https://github.com/user-attachments/assets/140a6d67-9319-4a53-85ee-bf6790593961" />

lindsaym-fa committed:
Revise `classic` palette, closes #707 (#716)

* Change ✨ everything ✨

* No changes to `gray`

LeaVerou committed:
Add accent color tint as separate variable (#725)

Co-authored-by: lindsaym-fa <dev@lindsaym.design>

LeaVerou commented:
Okay, I fixed the issues on Vercel. Phew, that was …not fun to track down. Turns out Vercel does some HTML-fu which produced a different result in a certain edge case.

Should be ready for another look!

I also added a prototype of the brand color picker.

LeaVerou committed:
Update remix.js

LeaVerou committed:
Add brand color metadata

LeaVerou committed:
Prototype of accent color picker

LeaVerou committed:
Display brand color in docs

lindsaym-fa commented:
Working much better! A couple of more minor but still buggy things:

1) When I open the Remix details, either the values in Palette and Brand Color are missing, or the Brand Color swatch is missing. I've never seen all of the fields render exactly as intended on initial open.
<img width="1023" alt="Screenshot 2025-02-07 at 10 18 09 AM" src="https://github.com/user-attachments/assets/b8387220-7cc8-469f-be00-7df21baba180" />
<img width="1023" alt="Screenshot 2025-02-07 at 10 20 24 AM" src="https://github.com/user-attachments/assets/285243de-c10a-44d5-86fb-a08795c20207" />

---

2) The theme default swatch in the Brand Color dropdown doesn't render quite right in many cases.
<img width="244" alt="Screenshot 2025-02-07 at 10 20 50 AM" src="https://github.com/user-attachments/assets/e632f9df-f966-45fb-a569-c36eccd746db" />
<img width="240" alt="Screenshot 2025-02-07 at 10 25 02 AM" src="https://github.com/user-attachments/assets/ce700733-f7f1-4134-a8f6-f3c4698d4c4f" />
<img width="240" alt="Screenshot 2025-02-07 at 10 22 00 AM" src="https://github.com/user-attachments/assets/948f1d03-ee8b-44aa-9d4e-cf4bba4ac3f1" />

The first example tends to occur when you open the Brand Color field first without manipulating any others.
The second example tends to occur when you change the Colors field, then open the Brand Color field only for themes whose default brand color is more than four characters (Active with Green, Tailspin with Indigo, Playful with Purple, etc.).
The last example happens for some themes when  both the Colors and Palette fields are manipulated. This seems to be due to the fields changing widths when selections are made, e.g.:

https://github.com/user-attachments/assets/f71d3ad7-4e62-4d8d-b072-14bcc1ba856f

https://github.com/user-attachments/assets/a39532b2-1ea3-40c6-a4d1-6365badf1faf

Usually, once I specifically select a Brand Color, the field width resets and the swatches render as expected.

LeaVerou committed:
Fix bugs with brand color

LeaVerou commented:
Can you check again? I just pushed a commit that (I think) fixes all of these issues!

LeaVerou committed:
Fix shrinking icon

LeaVerou commented:
Additionally, the brand color picker now displays the colors in the actual palette selected:

<img width="251" alt="image" src="https://github.com/user-attachments/assets/85bd496e-381f-45f8-b284-8d817a3d3544" />

I also improve the color section for the theme being viewed:

<img width="1052" alt="image" src="https://github.com/user-attachments/assets/683e494a-f264-46a7-9339-c653650c623b" />

lindsaym-fa commented:
The first issue is resolved 🎉, though the second is still present.

<img width="241" alt="Screenshot 2025-02-07 at 11 13 35 AM" src="https://github.com/user-attachments/assets/3bb47bfa-731b-4f48-b4b4-c3ecc09c7efd" />
<img width="235" alt="Screenshot 2025-02-07 at 11 14 06 AM" src="https://github.com/user-attachments/assets/3f4e0bc7-379b-4c25-8817-8e97ec163851" />
<img width="227" alt="Screenshot 2025-02-07 at 11 14 38 AM" src="https://github.com/user-attachments/assets/05e9bf52-123d-45a9-920f-1fa4e3aef349" />

LeaVerou commented:
Can you try again? I think you may have pulled too soon :)

lindsaym-fa reviewed (approved):
Looks good!
